File: common/interfaces/game-info.ts
import { GameVisibilities } from '@common/enums/game-visibilities';
import { CommonGameConfig2 } from '@common/interfaces/lobby/common-game-config2';
import { User } from '@common/interfaces/user/user';

export interface Game {
    gameConfig: CommonGameConfig2;
    visibility: GameVisibilities;
    lobbyId: string;
    players: User[];
    observers: User[];
    virtualPlayerNames: string[];
    isOngoing: boolean;
}

export interface GameClassic extends Game {}

export interface GameTournament extends Game {}

export interface GameCooperative extends Game {}

File: mobile/lib/data/models/interfaces/lobby/game_info.dart
import 'package:Scrabble/data/models/interfaces/lobby/common_game_config.dart';
import 'package:Scrabble/data/models/interfaces/serializable.dart';
import 'package:Scrabble/data/models/interfaces/user/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'game_info.g.dart';

@JsonSerializable(explicitToJson: true)
class GameInfo extends Serializable<GameInfo> {
  final CommonGameConfig gameConfig;
  final String visibility;
  final String lobbyId;
  final List<User> players;
  final List<User> observers;
  final List<String> virtualPlayerNames;
  bool isOngoing;

  GameInfo(this.gameConfig, this.visibility, this.lobbyId, this.players, this.observers, this.virtualPlayerNames,
      this.isOngoing);

  @override
  List<Object> get props => [gameConfig, lobbyId, players, observers, virtualPlayerNames];

  factory GameInfo.fromJson(Map<String, dynamic> json) => _$GameInfoFromJson(json);

  @override
  GameInfo fromJson(Map<String, dynamic> json) => _$GameInfoFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$GameInfoToJson(this);
}
